pkg/reconciler/labeler/labels.go

```diff
@@ -53,9 +53,22 @@ func (c *Reconciler) syncLabels(ctx context.Context, r *v1.Route) error {
 			return err
 		}
 		revisions.Insert(tt.RevisionName)
+
+		// If the owner reference is a configuration, add it to the list of configurations
+		// and add its LatestCreatedRevision to the list of revisions
 		owner := metav1.GetControllerOf(rev)
 		if owner != nil && owner.Kind == "Configuration" {
 			configs.Insert(owner.Name)
+
+			config, err := c.configurationLister.Configurations(r.Namespace).Get(owner.Name)
+			if err != nil {
+				return err
+			}
+
+			// If the configuration is being updated, the route's .status.traffic will
+			// only incude the latest ready revision. We need to label the latest created revision
+			// as well so that there is a smooth transition when the new revision becomes ready.
+			revisions.Insert(config.Status.LatestCreatedRevisionName)
```

Comment:
Perhaps we should only do this for Configurations that are named in the Route's `.spec.traffic`?  If folks are using `revisionName` then this could keep a `minScale: N` revision from scaling to zero.

Reply:
It looks like the reconciler is actually labelling _all_ configurations, not just configurations for `LatestRevision` revisions. 

I used the `LatestRevision` field in `TrafficTarget` to determine if a configuration should be labeled, and in that case label the `LatestCreatedRevision` as well. WDYT?

Reply:
Right, that just means we'll get more enqueues due to Configuration changes than we need.   A superset is much better than a subset! 😉 

I think we want to keep labeling all Configurations, not just those referenced explicitly, and then we should be keying off of `spec.traffic`'s `configurationName` references for the LCR to label.

Reply:
> I think we want to keep labeling all Configurations, not just those referenced explicitly, and then we should be keying off of `spec.traffic`'s `configurationName` references for the LCR to label.

If I understand correctly, this is to cover configurations in `spec.traffic` that don't have ready revisions and thus won't show up in `status.traffic`?

Reply:
From slack:
> We want the configuration labeled whenever it or one of its revisions are in a route